src/armor_layers.cpp

```diff
@@ -112,387 +106,499 @@ item_penalties get_item_penalties( std::list<item>::const_iterator worn_item_it,
+                } else {
+                    agg[i] += get_col_value( static_cast<sort_armor_columns>( i ) ) * get_col_value( column_coverage );
+                }
+            }
+        }
 };
 
-static std::vector<layering_item_info> items_cover_bp( const Character &c, int bp )
+class layering_group_info
 {
-    std::vector<layering_item_info> s;
-    for( auto elem_it = c.worn.begin(); elem_it != c.worn.end(); ++elem_it ) {
-        if( elem_it->covers( static_cast<body_part>( bp ) ) ) {
-            s.push_back( { get_item_penalties( elem_it, c, bp ),
-                           elem_it->get_encumber( c ),
-                           elem_it->tname()
-                         } );
+    private:
+        std::string name;
+        std::array<int, column_num_entries> header_data;
+        std::list<layering_item_info> items;
+        body_part bp;
+        player *p;
+        int start_line;
+
+        int divider_posn;
+
+        //printing constatnts
+        int x_bound;
+        int max_y;
+        //printing vars
+        int list_min_line;
+        int list_selected_line;
+        bool is_moving;
+
+    public:
+        bool is_skipped = false;
+
+        layering_group_info( body_part bp, player *p ) {
+            this->bp = bp;
+            this->p = p;
+        }
+        void add_item( std::list<item>::iterator itm_i ) {
+            items.push_back( layering_item_info( itm_i, *p, bp ) );
+        }
+        // same for the duration of the menu
+        void set_print_constats( int right_bound, int max_y ) {
+            this->x_bound = right_bound;
+            this->max_y = max_y;
+        }
+        // depend on user input
+        void set_print_vars( int list_selected_line, bool is_moving, int list_min_line ) {
+            this->list_selected_line = list_selected_line;
+            this->is_moving = is_moving;
+            this->list_min_line = list_min_line;
         }
-    }
-    return s;
-}
 
-static void draw_grid( const catacurses::window &w, int left_pane_w, int mid_pane_w )
-{
-    const int win_w = getmaxx( w );
-    const int win_h = getmaxy( w );
-
-    draw_border( w );
-    mvwhline( w, point( 1, 2 ), 0, win_w - 2 );
-    mvwvline( w, point( left_pane_w + 1, 3 ), 0, win_h - 4 );
-    mvwvline( w, point( left_pane_w + mid_pane_w + 2, 3 ), 0, win_h - 4 );
-
-    // intersections
-    mvwputch( w, point( 0, 2 ), BORDER_COLOR, LINE_XXXO );
-    mvwputch( w, point( win_w - 1, 2 ), BORDER_COLOR, LINE_XOXX );
-    mvwputch( w, point( left_pane_w + 1, 2 ), BORDER_COLOR, LINE_OXXX );
-    mvwputch( w, point( left_pane_w + 1, win_h - 1 ), BORDER_COLOR, LINE_XXOX );
-    mvwputch( w, point( left_pane_w + mid_pane_w + 2, 2 ), BORDER_COLOR, LINE_OXXX );
-    mvwputch( w, point( left_pane_w + mid_pane_w + 2, win_h - 1 ), BORDER_COLOR, LINE_XXOX );
-
-    wrefresh( w );
-}
+        void set_combined( bool combined ) {
+            this->name = body_part_name_as_heading( all_body_parts[bp], combined ? 2 : 1 );
```

Comment:
```suggestion
            name = body_part_name_as_heading( all_body_parts[bp], combined ? 2 : 1 );
```
`this` is usually redundant and should not be used in that case (simply because it does not add anything useful).